### ISSUE TITLE
shape[0], or shape[-1] or len on 1d-arrays

### ISSUE THREAD

**Issue by dneise** [cta-sst-1m/digicamtoy#8]
Sometimes one does not know, when reading code what the dimension of an array might be.

Could be 2d .. could be 1d ... 

Using `shape[-1]` or `shape[0]` implies the thing has more than one dimension, since one clearly wants to take a special element from the shape. 

I think using `len` implies the thing has only one dimension. So using `shape[0]` for a 1d array confuses me a lot.

I think it is event more confusing, when `shape[0]` and `shape[-1]` are both used on a 1d-array in the same code in different places. Since it looks like the author wants to explicitly express a difference between these numbers, while in reality it is exactly the same number.

**Comment by dneise** [comment]
Here is an example:

https://github.com/cta-sst-1m/digicamtoy/blob/8c5c3df1734375ac9bba9bad35a59c9a17bb765d/digicamtoy/tracegenerator/__init__.py#L163-L165

sampling_bins is 1d ... in line 165 `shape[0]` is used on it.

https://github.com/cta-sst-1m/digicamtoy/blob/8c5c3df1734375ac9bba9bad35a59c9a17bb765d/digicamtoy/tracegenerator/__init__.py#L316-L319

and in line 318 `shape[-1]` is used on it

**Comment by calispac** [comment]
I agree it can be confusing. I think shape[-1] is always better than shape[0] (more stable if you add an extradimension). But you can go ahead with what you prefer here

**Comment by dneise** [comment]
I do not agree that this is more stable ... consider this example:
```python
In [1]: a.shape, b.shape
Out[1]: ((3, 4), (3, 4))

In [2]: c = np.dstack([a,b]); c.shape
Out[2]: (3, 4, 2)
```

In this case `a.shape[-1]` returns `4` which might be "number of samples" and `c.shape[-1]` returns `2` which might be "number of gain channels or so" ... 

So here stability would be, to state explicitly which axis you mean like `a.shape[1] == c.shape[1]`.

It really depends on context I think...

**Comment by calispac** [comment]
Sure but in CTA / `ctapipe` waveforms have shapes : 
`(n_gain, n_pixels, n_samples)` or `(n_pixels, n_samples)`.

> It really depends on context I think...

Yes
